### ISSUE TITLE
Connection Link for standard connectors causes panic

### ISSUE THREAD

**Issue by dbryar** [pingidentity/terraform-provider-davinci#289]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### DaVinci Terraform provider Version

0.3.0

### Terraform Version

1.4.2

### Affected Resource(s)

- `davinci_connection`
- `davinci_flow`

### Terraform Configuration Files
in a `davinci_flow` block;
```hcl
  connection_link {
    id   = "${data.davinci_connection.teleport.id}"
    name = "${data.davinci_connection.teleport.name}"
  }
```

### Debug Output

various attempts to get this running but I posted the panic output in a previous bug thinking it was related.

It seems the provider is forcing the connection link to have both a name and an ID but the default connections crash on `plan` when both are included (as shown), and it throws an error is one is missing.

### Panic Output

```
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ValidateResourceConfig call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-davinci_v0.3.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xc378ca]
goroutine 278 [running]:
github.com/pingidentity/terraform-provider-davinci/internal/service/davinci.validateConnectionSubflowLinkMappings({0xf84548, 0xc000b7ce40}, {{0x2, {0xc00133aa80, 0x633d}}, {0x0, 0x0, 0x1, 0x0, 0x1, ...}}, ...)
	github.com/pingidentity/terraform-provider-davinci/internal/service/davinci/resource_flow.go:1233 +0x10ea
github.com/pingidentity/terraform-provider-davinci/internal/service/davinci.(*FlowResource).ValidateConfig(0xc000b7ce40?, {0xf84548, 0xc000b7ce40}, {{{{0xf891c0, 0xc0013486c0}, {0xcfb400, 0xc0013338c0}}, {0xf8a918, 0xc0002add60}}}, 0xc00[130](https://gitlab.com/b707/platform-and-payments/pinocchio/-/jobs/6605732925#L130)6b28)
	github.com/pingidentity/terraform-provider-davinci/internal/service/davinci/resource_flow.go:505 +0x1dd
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ValidateResourceConfig(0xc000467d40, {0xf84548, 0xc000b7ce40}, 0xc0013061c8, 0xc001357650)
	github.com/hashicorp/terraform-plugin-framework@v1.5.0/internal/fwserver/server_validateresourceconfig.go:93 +0x44b
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ValidateResourceConfig(0xc000467d40, {0xf84548?, 0xc000b7cd20?}, 0xc001306150)
	github.com/hashicorp/terraform-plugin-framework@v1.5.0/internal/proto6server/server_validateresourceconfig.go:47 +0x47d
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ValidateResourceConfig(0xf84580?, {0xf84548?, 0xc000b7ca20?}, 0xc001306150)
	github.com/hashicorp/terraform-plugin-mux@v0.14.0/tf6muxserver/mux_server_ValidateResourceConfig.go:36 +0x192
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ValidateResourceConfig(0xc00065f4a0, {0xf84548?, 0xc000b7c2a0?}, 0xc001302080)
	github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/tf6server/server.go:716 +0x2eb
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ValidateResourceConfig_Handler({0xde8040?, 0xc00065f4a0}, {0xf84548, 0xc000b7c2a0}, 0xc00[131](https://gitlab.com/b707/platform-and-payments/pinocchio/-/jobs/6605732925#L131)e080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:410 +0x16c
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000637e00, {0xf84548, 0xc000b7c1e0}, {0xf89ad0, 0xc0005031e0}, 0xc001308000, 0xc0004138c0, 0x156b368, 0x0)
	google.golang.org/grpc@v1.61.0/server.go:1385 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc000637e00, {0xf89ad0, 0xc0005031e0}, 0xc001308000)
	google.golang.org/grpc@v1.61.0/server.go:1796 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 40
	google.golang.org/grpc@v1.61.0/server.go:1040 +0x[135](https://gitlab.com/b707/platform-and-payments/pinocchio/-/jobs/6605732925#L135)
Error: The terraform-provider-davinci_v0.3.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

standard connectors should only need an ID or a name (per docs)

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

can't link standard connectors from `data` source, only from `resource` when configured

### References

See error #230

**Comment by dbryar** [comment]
From !230

**Comment by patrickcping** [comment]
Hey @dbryar , thanks for raising!

This is resolved in https://github.com/pingidentity/terraform-provider-davinci/pull/285 , the release that includes this fix will be issued today

**Comment by patrickcping** [comment]
Just released in `v0.3.1`
